src/Button/Button.js

```diff
@@ -118,15 +125,33 @@ export const Button = React.forwardRef(
         ref={ref}
         {...props}
       >
+        {icon && iconPosition === 'left' ? (
+          <HeroIcon
+            icon={heroicon}
+            aria-hidden
+            className={dcnb(
+              'su-inline-block',
+              levers.animate,
+              levers.iconPositionStyles,
+              iconClasses
+            )}
+            {...iProps}
+          />
+        ) : null}
         {children}
-        {icon && (
+        {icon && iconPosition === 'right' ? (
           <HeroIcon
             icon={heroicon}
             aria-hidden
-            className={dcnb('su-inline-block', levers.animate, iconClasses)}
+            className={dcnb(
+              'su-inline-block',
+              levers.animate,
+              levers.iconPositionStyles,
+              iconClasses
+            )}
             {...iProps}
           />
-        )}
+        ) : null}
```

Comment:
Is it possible to place the children inside the conditional render blocks instead of the HeroIcon? That way there's a bit less duplicate code 😄

Reply:
Same for the other 2 components ^^^